src/firefly/js/core/LayoutCntlr.js

```diff
@@ -316,6 +317,14 @@ export function dropDownHandler(layoutInfo, action) {
         case TBL_RESULTS_ADDED:
         case REPLACE_VIEWER_ITEMS :
         case ImagePlotCntlr.PLOT_IMAGE :
+            if (action.type === TBL_RESULTS_ADDED || action.type ===  TABLE_LOADED
+                || action.type === TBL_RESULTS_ACTIVE || action.type === CHART_ADD) {
+                //TODO: Review: is it okay to get tblId from gropupId when action.type is CHART_ADD?
+                const tbl = action.type === CHART_ADD ? getTblById(action.payload.groupId) : getTblById(action.payload.tbl_id);
+                if (tbl?.request?.META_INFO?.UploadTable) {
+                    return layoutInfo;
+                }
```

Comment:
this code is not quite right. I think you should clean it up. You have an `if` inside a `case` which mostly should not happen. it should be another case. Also, some of the if `action.type === TBL_RESULTS_ACTIVE` and `action.type ===  TABLE_LOADED` can never be true because they are not part of the case.